### PR TITLE
Introduce Freshdesk-Migrator documentation

### PR DESCRIPTION
**Pull request by MrGeneration**
This PR introduces the freshdesk migrator that has been added with Zammad 4.1.

### CODE REVIEW

File: migration/freshdesk.rst [hunk 1]
```diff
@@ -0,0 +1,274 @@
+from Freshdesk
```
Comment (YetAnotherGerrit) [inline migration/freshdesk.rst:1]: From Freshdesk

File: migration/freshdesk.rst [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+Please note below Freshdesk specific limitations. 
+These are additional limitations to the
+:ref:`general ones listed <migration_limitations>`.
+
+   * | Differential migrations are **not** supported!
+     | The general suggestion is to run a test import before to learn
+       how long the migration is going to take.
+   * **Important:** Please note that migration speed highly depends on your
+     Freshdesk plan (API rate limits apply)
```
Comment (YetAnotherGerrit) [inline migration/freshdesk.rst:15]: Dot at the end of the sentence.

File: migration/freshdesk.rst [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+   * Due to API limitations Zammad will not show the total number of objects
+     to import, but instead correct them in steps of ``100``.
+   * User passwords are not migrated and will require password resets.
```
Comment (YetAnotherGerrit) [inline migration/freshdesk.rst:18]: How do I reset all passwords? Link or elaboration would be helpful here.

File: migration/freshdesk.rst [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+Prerequisites
+=============
+
+Zammad requires API access which is why you'll need to `create an API key`_
+for the migration. The migrator will request your Freshdesk subdomain and
+API key.
+
+.. warning:: **🥸 To be or not to be**
+
+  Ensure to retrieve the API key with an **full administrator** account.
```
Comment (YetAnotherGerrit) [inline migration/freshdesk.rst:34]: "a" full account.

File: migration/freshdesk.rst [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+Importing Freshdesk data
+========================
+
+Generally you have two options on how to migrate data.
+If you have a fairly big instance with a lot of data, you may want to
+consider using the console over the browser version.
+
+.. tabs::
+
+   .. tab:: via Browser
```
Comment (YetAnotherGerrit) [inline migration/freshdesk.rst:49]: "Via browser", not "via Browser".

File: migration/freshdesk.rst [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+
+      After installing Zammad and configuring your 
+      :doc:`webserver </getting-started/configure-webserver>`, navigate to your 
+      Zammads FQDN in your Browser and follow the migration wizard.
```
Comment (YetAnotherGerrit) [inline migration/freshdesk.rst:53]: "browser", not "Browser".

File: migration/freshdesk.rst [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+         .. figure:: /images/migration/freshdesk/scheduler-interrupted.png
+
+         If this message appears after providing your credentials, please be
+         patient. The migration should start within 5 minutes.
+
+         If you receive above message after the migration begun, please
+         consider using the console approach instead and reset the
+         installation.
+
+   .. tab:: via Console
```
Comment (YetAnotherGerrit) [inline migration/freshdesk.rst:72]: "Via console", not "via Console".

**Review by YetAnotherGerrit — CHANGES_REQUESTED** [review]

File: migration/freshdesk.rst (hunk 3 above)
Comment (MrGeneration) [reply migration/freshdesk.rst:18]: You can't. I would never -ever- suggest a default password for all user accounts.
Do you think expecting the users to reset their passwords is a bad behavior?

Passwords that are known to the admin are -per definition- unsafe.

Comment (YetAnotherGerrit) [reply migration/freshdesk.rst:18]: I thought more of a function that sends password reset mails to all agents and admins. But as I learned, there is no function like that, I suggest the following sentence:
User passwords are not migrated and will require the users to use the password reset link on the login page.

Comment (MrGeneration) [reply migration/freshdesk.rst]: Okay. I also added a link to the security section of the admin documentation as I felt that is absolutely required for context.

**Pull request merged**